### PR TITLE
GPU Query infrastructure

### PR DESCRIPTION
**Pull request by venkatn93**
This PR provides the infrastructure for GL to be queried for various stats of the GPU like time elapsed to draw a frame, number of primitives drawn, etc. This is an early draft that has only been tested for time elapsed but can be used for various other queries to find out the state of the GPU.
It s a work in a progress that is not usable yet.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5ededf75157ad0a80754f8a9561547de84b81bbe available at: https://gist.github.com/c73756919e3bb458c6a2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5277.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5277.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a6ec668b2e5b116eaa0e82dbe5375ee141817a70 available at: https://gist.github.com/c73756919e3bb458c6a2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5277.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5277.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 250d2e9e2c061916b5554729febc827eaea3e51a available at: https://gist.github.com/c73756919e3bb458c6a2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5277.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5277.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4478d76618efb2b7ada6a2444e4990490639c754 available at: https://gist.github.com/c73756919e3bb458c6a2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5277.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5277.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1894b42773745e8f46e142aef3ee397f82cd034d available at: https://gist.github.com/c73756919e3bb458c6a2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5277.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5277.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 58f127f29946f0ad0940219f0f2ba66842183973 available at: https://gist.github.com/c73756919e3bb458c6a2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5277.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5277.dmg

**Comment by ZappoMan** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 58f127f29946f0ad0940219f0f2ba66842183973 available at: https://gist.github.com/c73756919e3bb458c6a2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5277.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5277.dmg

File: libraries/gpu/src/gpu/GLBackendQuery.cpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    }
+}
+
+void GLBackend::do_getQuery(Batch& batch, uint32 paramOffset) {
+    auto query = batch._queries.get(batch._params[paramOffset]._uint);
+    GLQuery* glquery = syncGPUObject(*query);
+    if (glquery) { 
+        #if (GPU_FEATURE_PROFILE == GPU_LEGACY)
+            // (EXT_TIMER_QUERY)
+            glGetQueryObjectui64vEXT(glquery->_qo, GL_QUERY_RESULT, &glquery->_result);
```
Comment (ZappoMan) [inline libraries/gpu/src/gpu/GLBackendQuery.cpp]: this is the source of the problem with the linux build. If you want to work around it for now, you could basically #ifndef around this function call on unix. You of course won't get your doQuery() on unix, but at least you'll get a windows and mac build.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2e5fd04d9e121cb5256d84e7d446169668c34a0b available at: https://gist.github.com/c73756919e3bb458c6a2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5277.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5277.dmg

**Comment by samcake** [comment]
Ready to Merge, please give  us a bit more details in the description of the PR
in particular the fact that it's a first drafty version that is not yet usable!

**Pull request merged**